core/src/test/scala/pool.scala

```diff
@@ -68,4 +72,70 @@ class PoolSuite extends FunSuite {
     assert(regCells.size === 1000)
   }
 
+  test("onQuiescent(cell): incomplete cell") {
+    val latch = new CountDownLatch(1)
+
+    val pool = new HandlerPool
+    val completer1 = CellCompleter[StringIntKey, Int]("somekey")(new StringIntUpdater, pool)
+
+    var i = 0
+    while (i < 10000) {
+      val p1 = Promise[Boolean]()
+      val p2 = Promise[Boolean]()
+      pool.execute { () => { p1.success(true) }: Unit }
+      pool.onQuiescent { () => p2.success(true) }
```

Comment:
The type ascription in the call to `execute` is not so nice. Can we get rid of it?